clients/lsp-rust.el

```diff
@@ -1289,6 +1290,134 @@ https://github.com/rust-analyzer/rust-analyzer/blob/master/docs/dev/lsp-extensio
+
+(defvar font-lock-library-modifier-face 'font-lock-nop-face
+  "Face name to use for library items.")
+
+(defvar font-lock-mutable-modifier-face 'font-lock-mutable-modifier-face
+  "Face name to use for mutable items.")
+
+(defvar font-lock-public-modifier-face 'font-lock-nop-face
+  "Face name to use for public items.")
+
+(defvar font-lock-reference-modifier-face 'font-lock-reference-modifier-face
+  "Face name to use for reference items.")
+
+(defvar font-lock-trait-modifier-face 'font-lock-nop-face
+  "Face name to use for trait items.")
+
+(defvar font-lock-unsafe-modifier-face 'font-lock-nop-face
+  "Face name to use for unsafe items.")
+
+(defface font-lock-mutable-modifier-face
+  '((t :underline t))
+  "Font Lock mode face used to highlight keywords."
+  :group 'font-lock-faces)
+
+(defface font-lock-reference-modifier-face
+  '((t :bold t))
+  "Font Lock mode face used to highlight keywords."
+  :group 'font-lock-faces)
+
+(defvar font-lock-nop-face 'font-lock-nop-face
+  "Placeholder face name that does not actually modify the face.")
+
+(defface font-lock-nop-face
+  '((t nil))
+  "Placeholder face name that does not actually modify the face."
+  :group 'font-lock-faces)
+
+
+(defun lsp-rust-analyzer--set-tokens ()
+  "Set the mapping between rust-analyzer keywords and fonts to apply.
+The keywords are sent in the initialize response, in the semantic
+tokens legend."
+  (setq lsp-semantic-token-modifier-faces
+        '(
+          ("documentation" . font-lock-documentation-modifier-face)
+          ("declaration" . font-lock-declaration-modifier-face)
+          ("definition" . font-lock-definition-modifier-face)
+          ("static" . font-lock-static-modifier-face)
+          ("abstract" . font-lock-abstract-modifier-face)
+          ("deprecated" . font-lock-deprecated-modifier-face)
+          ("readonly" . font-lock-readonly-modifier-face)
+          ("default_library" . font-lock-default-library-modifier-face)
+          ("async" . font-lock-async-modifier-face)
+          ("attribute" . font-lock-attribute-modifier-face)
+          ("callable" . font-lock-callable-modifier-face)
+          ("constant" . font-lock-constant-modifier-face)
+          ("consuming" . font-lock-consuming-modifier-face)
+          ("control_flow" . font-lock-control-flow-modifier-face)
+          ("crate_root" . font-lock-crate-root-modifier-face)
+          ("injected" . font-lock-injected-modifier-face)
+          ("intra_doc_link" . font-lock-intra-doc-link-modifier-face)
+          ("library" . font-lock-library-modifier-face)
+          ("mutable" . font-lock-mutable-modifier-face)
+          ("public" . font-lock-public-modifier-face)
+          ("reference" . font-lock-reference-modifier-face)
+          ("trait" . font-lock-trait-modifier-face)
+          ("unsafe" . font-lock-unsafe-modifier-face)
+          )))
+
+(with-eval-after-load 'rust-mode
+  (when lsp-semantic-tokens-enable
+    (lsp-rust-analyzer--set-tokens)
```

Comment:
I don't think this call should be here. Technically, `(require 'rust-mode)` can happen from any buffer and if you want to keep this configuration only for rust buffer then you can use :after-open-fn like this: https://github.com/yyoncho/lsp-mode/blob/master/clients/lsp-go.el#L330

Reply:
I should fix this in the terraform client too.